### DISCUSSION TITLE
Restructure Package Naming

### DISCUSSION THREAD

**Discussion by juliapampus** [eclipse-edc/Connector#1924] [Ideas]
# Package Renaming

If you take a look at the current packaging structure, the naming is not consistent. Sometimes it is \*.core and core.\*, same goes for spi. The module refactoring did not make it better. As people are using the EDC artifacts as dependecy, we need to discuss about a cleaner package naming.

Initial proposal for the "new" column tbw.

## :core
| module | current package naming | new
| :--- | :--- | :---
| :common:base |   *.core  |   *.connector.
| :common:boot |   *.boot  |   *.connector.
| :common:jwt-core |   *.core.jwt  |   *.connector.
| :common:policy-engine    |   *.policy.engine  |   *.connector.
| :common:policy-evaluator |   *.policy.model  |   *.connector.
| :common:state-machine    |   *.common.statemachine  |   *.connector.
| :common:util |   *.common  |   *.connector.
| :control-plane:contract  |   *.contract  |   *.connector.
| :control-plane:control-plane-core    |   *.core.controlplane  |   *.connector.
| :control-plane:transfer  |   *.transfer.core  |   *.connector.
| :data-plane:data-plane-core |   -  |   -
| :data-plane:data-plane-framework |   *.dataplane.framework  |   *.connector.
| :data-plane-selector:data-plane-selector-core    |   *.dataplane.selector  |   *.connector.
| :federated-catalog:federated-catalog-core    |   *.catalog  |   *.connector.

## :data-protocols
| module | current package naming | new
| :--- | :--- | :---
| :ids:ids-api-configuration |   *.ids.api.configuration  |   *.connector.
| :ids:ids-api-multipart-dispatcher-v1   |   *.ids.api.multipart.dispatcher  |   *.connector.
| :ids:ids-api-multipart-endpoint-v1 |   *.ids.api.multipart  |   *.connector.
| :ids:ids-core  |   *.ids.core  |   *.connector.
| :ids:ids-jsonld-serdes |   *.ids.jsonld  |   *.connector.
| :ids:ids-spi   |   *.ids.spi  |   *.connector.
| :ids:ids-token-validation  |   *.ids.token.validation  |   *.connector.
| :ids:ids-transform-v1  |   *.ids.transform  |   *.connector.

## :extensions
| module | current package naming | new
| :--- | :--- | :---
| :common:api:api-core   |   *.api  |   *.connector.
| :common:api:observability  |   *.api.observability  |   *.connector.
| :common:auth:auth-basic    |   *.api.auth  |   *.connector.
| :common:auth:auth-tokenbased   |   *.api.auth  |   *.connector.
| :common:aws:aws-test   |   *.aws.testfixtures  |   *.connector.
| :common:aws:s3-core    |   *.aws.s3.core  |   *.connector.
| :common:azure:azure-eventgrid  |   *.events.azure  |   *.connector.
| :common:azure:azure-resource-manager   |   *.azure.resourcemanager  |   *.connector.
| :common:azure:azure-test   |   *.azure.testfixtures  |   *.connector.
| :common:azure:blob-core    |   *.azure.blob.core  |   *.connector.
| :common:azure:cosmos-common    |   *.azure.cosmos  |   *.connector.
| :common:configuration:filesystem-configuration |   *.configuration.fs  |   *.connector.
| :common:events:cloudevents-http    |   *.events.cloudevents.http  |   *.connector.
| :common:http   |   -  |   -
| :common:http:jersey    |   *.extension.jersey  |   *.connector.
| :common:http:jersey-micrometer |   *.extension.jersey  |   *.connector.
| :common:http:jetty |   *.extension.jetty  |   *.connector.
| :common:http:jetty-micrometer  |   *.extension.jetty  |   *.connector.
| :common:iam:decentralized-identity |   -  |   -
| :common:iam:decentralized-identity:dummy-credentials-verifier  |   *.iam.verifier  |   *.connector.
| :common:iam:decentralized-identity:identity-common-test    |   *.iam.did.testfixtures  |   *.connector.
| :common:iam:decentralized-identity:identity-did-core   |   *.iam.did  |   *.connector.
| :common:iam:decentralized-identity:identity-did-crypto |   *.iam.did.crypto  |   *.connector.
| :common:iam:decentralized-identity:identity-did-service    |   *.identity  |   *.connector.
| :common:iam:decentralized-identity:identity-did-web    |   *.iam.did.web  |   *.connector.
| :common:iam:iam-mock   |   *.iam.mock  |   *.connector.
| :common:iam:oauth2:daps    |   *.iam.daps  |   *.connector.
| :common:iam:oauth2:oauth2-core |   *.iam.oauth2.core  |   *.connector.
| :common:junit  |   *.junit  |   *.connector.
| :common:micrometer |   *.metrics.micrometer  |   *.connector.
| :common:monitor:jdk-logger-monitor |   *.logger  |   *.connector.
| :common:sql:common-sql |   *.sql  |   *.connector.
| :common:sql:lease-sql  |   *.sql.lease  |   *.connector.
| :common:sql:pool:apache-commons-pool-sql   |   *.sql.pool.commons  |   *.connector.
| :common:transaction    |   -  |   -
| :common:transaction:transaction-atomikos   |   *.transaction.atomikos  |   *.connector.
| :common:transaction:transaction-local  |   *.transaction.local  |   *.connector.
| :common:vault:azure-vault  |   *.core.security.azure  |   *.connector.
| :common:vault:filesystem-vault |   *.core.security.fs |   *.connector.
| :common:vault:hashicorp-vault  |   *.core.security.hashicorpvault  |   *.connector.
| :control-plane:api:data-management |   -  |   -
| :control-plane:api:data-management:api-configuration   |   *.api.datamanagement.configuration  |   *.connector.
| :control-plane:api:data-management:asset-api   |   *.api.datamanagement.asset  |   *.connector.
| :control-plane:api:data-management:catalog-api |   *.api.datamanagement.catalog  |   *.connector.
| :control-plane:api:data-management:contractagreement-api   |   *.api.datamanagement.contractagreement  |   *.connector.
| :control-plane:api:data-management:contractdefinition-api  |   *.api.datamanagement.contractdefinition  |   *.connector.
| :control-plane:api:data-management:contractnegotiation-api |   *.api.datamanagement.contractnegotiation  |   *.connector.
| :control-plane:api:data-management:policydefinition-api    |   *.api.datamanagement.policy  |   *.connector.
| :control-plane:api:data-management:transferprocess-api |   *.api.datamanagement.transferprocess  |   *.connector.
| :control-plane:data-plane-transfer:data-plane-transfer-client  |   *.transfer.dataplane  |   *.connector.
| :control-plane:data-plane-transfer:data-plane-transfer-sync    |   *.transfer.dataplane.sync  |   *.connector.
| :control-plane:http-receiver   |   *.receiver.http  |   *.connector.
| :control-plane:provision:blob-provision    |   *.provision.azure  |   *.connector.
| :control-plane:provision:http-provisioner  |   *.transfer.provision.http  |   *.connector.
| :control-plane:provision:s3-provision  |   *.aws.s3.provision  |   *.connector.
| :control-plane:store:cosmos:assetindex-cosmos  |   *.assetindex.azure  |   *.connector.
| :control-plane:store:cosmos:contract-definition-store-cosmos   |   *.contract.definition.store  |   *.connector.
|                                                                           |   *.cosmos.policy.store.model  |   *.connector.
| :control-plane:store:cosmos:contract-negotiation-store-cosmos  |   *.contract.negotiation.store  |   *.connector.
| :control-plane:store:cosmos:control-plane-cosmos   |   -  |   -
| :control-plane:store:cosmos:policy-store-cosmos    |   *.cosmos.policy.store  |   *.connector.
| :control-plane:store:cosmos:transfer-process-store-cosmos  |   *.transfer.store.cosmos  |   *.connector.
| :control-plane:store:sql:asset-index-sql   |   *.sql.assetindex  |   *.connector.
| :control-plane:store:sql:contract-definition-store-sql |   *.sql.contractdefinition.store  |   *.connector.
| :control-plane:store:sql:contract-negotiation-store-sql    |   *.sql.contractnegotiation  |   *.connector.
| :control-plane:store:sql:control-plane-sql |   -  |   -
| :control-plane:store:sql:policy-store-sql  |   *.sql.policy  |   *.connector.
| :control-plane:store:sql:transfer-process-store-sql    |   *.sql.transferprocess  |   *.connector.
| :data-plane:data-plane-api |   *.dataplane.api  |   *.connector.
| :data-plane:data-plane-azure-storage   |   *.azure.dataplane.azurestorage  |   *.connector.
| :data-plane:data-plane-data-factory    |   *.azure.dataplane.azurefactory  |   *.connector.
| :data-plane:data-plane-http    |   *.dataplane.http  |   *.connector.
| :data-plane:data-plane-s3  |   *.aws.dataplane.s3  |   *.connector.
| :data-plane:integration-tests  |   *.dataplane.http  |   *.connector.
| :data-plane-selector:selector-api  |   *.dataplane.selector  |   *.connector.
| :data-plane-selector:selector-client   |   *.dataplane.selector  |   *.connector.
| :federated-catalog:store:fcc-node-directory-cosmos |   *.catalog.node.directory.azure  |   *.connector.

## :samples
| module | current package naming | new
| :--- | :--- | :---
| :01-basic-connector   |   -  |   -
| :02-health-endpoint   |   *.extensions.health  |   *.connector.
| :03-configuration |   *.extensions.health  |   *.connector.
| :04.0-file-transfer:consumer  |   -  |   -
| :04.0-file-transfer:integration-tests |   *.extension.sample.test  |   *.connector.
| :04.0-file-transfer:provider  |   -  |   -
| :04.0-file-transfer:transfer-file |   *.extensions.api  |   *.connector.
| :04.1-file-transfer-listener:consumer |   -  |   -
| :04.1-file-transfer-listener:file-transfer-listener-integration-tests |   *.extension.sample.test  |   *.connector.
| :04.1-file-transfer-listener:listener |   *.extensions.listener  |   *.connector.
| :04.2-modify-transferprocess:api |   *.extensions.api  |   *.connector.
| :04.2-modify-transferprocess:consumer |   -  |   -
| :04.2-modify-transferprocess:modify-transferprocess-sample-integration-tests  |   *.samples.sample042.test  |   *.connector.
| :04.2-modify-transferprocess:simulator    |   *.samples.sample042  |   *.connector.
| :04.2-modify-transferprocess:watchdog |   *.samples.sample042  |   *.connector.
| :04.3-open-telemetry:consumer |   -  |   -
| :04.3-open-telemetry:provider |   -  |   -
| :05-file-transfer-cloud:consumer  |   -  |   -
| :05-file-transfer-cloud:provider  |   -  |   -
| :05-file-transfer-cloud:transfer-file |   *.extensions.transfer  |   *.connector.
| :other:custom-runtime |   *.demo.runtime  |   *.connector.

## :spi
| module | current package naming | new
| :--- | :--- | :---
| :common:auth-spi  |   *.api.auth  |   *.connector.
| :common:catalog-spi   |   *.spi.types.domain.catalog  |   *.connector.
| :common:core-spi  |   *.spi  |   *.connector.
| :common:identity-did-spi  |   *.iam.did.spi  |   *.connector.
| :common:jwt-spi   |   *.spi.jwt  |   *.connector.
| :common:oauth2-spi    |   *.iam.oauth2.spi  |   *.connector.
| :common:transaction-datasource-spi    |   *.spi.transaction.datasource  |   *.connector.
| :common:transaction-spi   |   *.spi.transaction  |   *.connector.
| :common:transport-spi |   *.spi.transformer  |   *.connector.
| :common:web-spi   |   *.spi.exception  |   *.connector.
|                       |   *.spi.validation  |   *.connector.
| :control-plane:contract-spi   |   *.spi.contract  |   *.connector.
|                                   |   *.spi.types.domain  |   *.connector.
| :control-plane:control-plane-spi  |   -  |   -
| :control-plane:data-plane-transfer-spi    |   *.transfer.dataplane.spi  |   *.connector.
| :control-plane:policy-spi |   *.spi.policy.store  |   *.connector.
| :control-plane:transfer-spi   |   *.spi.transfer  |   *.connector.
|                                   |   *.spi.types.domain.transfer  |   *.connector.
| :data-plane:data-plane-spi    |   *.dataplane.spi  |   *.connector.
| :data-plane-selector:data-plane-selector-spi  |   *.dataplane.selector  |   *.connector.
| :federated-catalog:federated-catalog-spi  |   *.catalog.spi  |   *.connector.

## :system-tests
| module | current package naming | new
| :--- | :--- | :---
| :azure-data-factory-tests    |   *.system.tests.local  |   *.connector.
| :azure-tests |   *.system.tests.local  |   *.connector.
| :e2e-transfer-test:backend-service   |   *.test.e2e  |   *.connector.
| :e2e-transfer-test:control-plane |   -  |   -
| :e2e-transfer-test:control-plane-cosmosdb    |   *.controlplane  |   *.connector.
| :e2e-transfer-test:control-plane-postgresql  |   *.controlplane  |   *.connector.
| :e2e-transfer-test:data-plane    |   -  |   -
| :e2e-transfer-test:runner    |   *.test.e2e  |   *.connector.
| :runtimes:azure-data-factory-transfer-consumer   |   -  |   -
| :runtimes:azure-data-factory-transfer-provider   |   -  |   -
| :runtimes:azure-storage-transfer-consumer    |   -  |   -
| :runtimes:azure-storage-transfer-provider    |   -  |   -
| :runtimes:file-transfer-consumer |   -  |   -
| :runtimes:file-transfer-provider |   *.extensions.api  |   *.connector.
| :tests   |   *.system.tests  |   *.connector.

## :tooling
| module | current package naming | new
| :--- | :--- | :---
| :module-domain    |   *.tooling.module.domain  |   *.connector.
| :module-processor |   *.tooling.module.processor  |   *.connector.
| :module-processor-extension-test  |   *.tooling.sample.module  |   *.connector.
| :module-processor-spi-test    |   *.tooling.sample.module  |   *.connector.

**Comment by jimmarino** [comment]
Thanks for putting this together. Here are some things we need to consider:

1. We need to eliminate split packages. This should be done by defining a package naming convention for the unique "root" package of each jar produced.
2. Many of the core packages should not have "connector" in the package name since they are lower-level modules and shared across runtimes (e.g. Connector, Catalog services, Identity Hub, etc.)
3. I think all of the sample jars should be preceded with "sample.<root>"

Maybe we can start to distill the rules for package naming which will allow us to derive the individual module names?

**Reply by juliapampus** [reply]
I think it would be nice to have a decision record at the end. However, we can start a discussion here first, as there are multiple possible ways of package structuring. I'll try to draft an initial proposal for our meeting tomorrow.
